### PR TITLE
Fix escaping and edge cases

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

File: trillian/ctfe/handlers.go [hunk 1]
```diff
@@ -451,6 +450,10 @@ func getSTHConsistency(ctx context.Context, c LogContext, w http.ResponseWriter,
 
 		// We got a valid response from the server. Marshal it as JSON and return it to the client
 		jsonRsp.Consistency = rsp.Proof.Hashes
+		if jsonRsp.Consistency == nil {
+			// Force an empty slice to get '[]' rather than 'null' in the JSON.
+			jsonRsp.Consistency = make([][]byte, 0)
```
Comment (Martin2112) [inline trillian/ctfe/handlers.go]: What about having a preallocated empty slice that gets reused?

File: trillian/ctfe/handlers_test.go [hunk 2]
```diff
@@ -996,6 +1015,26 @@ func TestGetProofByHash(t *testing.T) {
 					},
 				},
 			},
+			httpRsp: &inclusionProof,
+		},
+		{
+			// Hash with URL-encoded %2B -> '+'.
+			req:  "hash=WtfX3Axbm7UwtY7GhHoAHPCtXJVrY5vZsH%2ByaXOD2GI=&tree_size=1",
```
Comment (Martin2112) [inline trillian/ctfe/handlers_test.go]: Github shows the '%' in red. Possibly it's confused by some encoding thing?

File: trillian/ctfe/handlers_test.go [hunk 3]
```diff
@@ -1140,7 +1206,7 @@ func TestGetSTHConsistency(t *testing.T) {
 			continue
 		}
 		if test.rpcRsp != nil || test.rpcErr != nil {
-			info.client.EXPECT().GetConsistencyProof(deadlineMatcher(), &trillian.GetConsistencyProofRequest{LogId: 0x42, FirstTreeSize: 10, SecondTreeSize: 20}).Return(test.rpcRsp, test.rpcErr)
+			info.client.EXPECT().GetConsistencyProof(deadlineMatcher(), gomock.Any()).Return(test.rpcRsp, test.rpcErr)
```
Comment (Martin2112) [inline trillian/ctfe/handlers_test.go]: Can we strengthen the gomock.Any? At least to the right type of object maybe?

File: trillian/ctfe/handlers.go (hunk 1 above)
Comment (daviddrysdale) [inline trillian/ctfe/handlers.go]: Done.

File: trillian/ctfe/handlers_test.go (hunk 2 above)
Comment (daviddrysdale) [inline trillian/ctfe/handlers_test.go]: I think it's supposed to be highlighting syntax errors, but % doesn't need to be escaped in a Go string so it looks like a false positive.

File: trillian/ctfe/handlers_test.go (hunk 3 above)
Comment (daviddrysdale) [inline trillian/ctfe/handlers_test.go]: Done.

**Comment by Martin2112** [comment]
LGTM

**Comment by RJPercival** [comment]
Fixes #32.

**Pull request merged**